test/staking/DonationsStaking.test.ts

```diff
@@ -309,9 +325,15 @@ describe("DonationsStaking - DonationStaking contract that receives funds in ETH
       "setStakingContract",
       [simpleStaking.address]
     );
+
     await genericCall(donationsStaking.address, encodedData);
+    const stakingAmountAfterSet = await goodCompoundStaking.balanceOf(
+      donationsStaking.address
+    );
     const stakingContractAfterSet = await donationsStaking.stakingContract();
     const stakingTokenAfterSet = await donationsStaking.stakingToken();
+    expect(stakingAmountBeforeSet).to.be.gt(0);
+    expect(stakingAmountAfterSet).to.be.equal(0);
```

Comment:
what are the actions that happen when setting a new contract?
then make sure you test all those actions